src/MoreVehicles/VehicleManagerPatch.cs

```diff
@@ -262,6 +265,33 @@ private static IEnumerable<CodeInstruction> Transform(IEnumerable<CodeInstructio
                 var replaced1024 = CodeProcessor.ReplaceOperands(instructions, 1024, 4096);
                 return CodeProcessor.ReplaceOperands(replaced1024, VanillaMaxVehicleCount, ModdedMaxVehicleCount);
             }
+        }
+
+        private sealed class VehicleManager_AirlineModified : PatchBase
+        {
+            protected override MethodInfo GetMethod() =>
+                typeof(VehicleManager).GetMethod(
+                    nameof(VehicleManager.AirlineModified),
+                    BindingFlags.Instance | BindingFlags.Public,
+                    null,
+                    new Type[0],
+                    new ParameterModifier[0]);
+
+            [System.Diagnostics.CodeAnalysis.SuppressMessage("Redundancy", "RCS1213", Justification = "Harmony patch")]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage("Naming Rules", "SA1313", Justification = "Harmony patch")]
+            private static bool Prefix(VehicleManager __instance)
+            {
+                for (int i = 0; i < __instance.m_vehicles.m_buffer.Length; ++i)

```

Comment:
This is a loop over the whole array (65k+ entries). I'd suggest to avoid doing this, especially in the game simulation loop. I don't know how often this method will be called on each simulation frame, but this might lead to a nasty performance drop.
Is there any way to fix the issue without a complete array scan?`This looks like a workaround to me.